### PR TITLE
Add Observer to select input device error

### PR DESCRIPTION
**Pull request by xuesichao**
**Issue #433 :** 

**Description of changes:**
Fix several typos. 
Add observer to `selectAudioInputDeviceError` and `selectVideoInputDeviceError` to deliver the error from SDK level to client application level.

When calling the `meetingManager.selectVideoInputDevice()/selectAudioInputDevice()`
* If there is an error, assign the error to `selectVideoInputDeviceError`/`selectAudioInputDeviceError`, and publish it
* If there is no error, set the `selectVideoInputDeviceError`/`selectAudioInputDeviceError` to `null`, and publish it

Error can be accessed through `useVideoInput`/`useAudioInput` hook. 

**Testing**
1. Have you successfully run `npm run build:release` locally?
    Yes

2. How did you test these changes?
    Mock a non-existent device for both `VideoInputProvider` and `AudioInputProvider`, and let `CameraSelection` and `MicSelection` display the `selectDeviceError` if not `null`, hide it if `null`.
    Run the meeting demo, on the "Device" page, for both video input device and audio input device:
    * Select the mocked device, and then the error appears
    * Choose the correct device, the error disappears
 
    Test on Safari, Chrome, Firefox.
3. If you made changes to the component library, have you provided corresponding documentation changes?
Yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,8 +17,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Added optional props to specify icon titles in the `AudioInputControl` and `ContentShareControl` components.
 - Added optional props to specify the dropdown text that shows when no video quality is selected, in the `QualitySelection` component.
-
+- Added observer to `selectAudioInputDeviceError` and `selectVideoInputDeviceError` to deliver the error from SDK level to client application level.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:20]: Could you please change this to "Add"? My bad that I didn't check the last two changelogs to follow this. But general good practice is to keep it in the present tense.

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # AudioInputProvider
 
-The `AudioInputProvider` provides a list of the user's available `audio-input` devices and the currently selected device.
+The `AudioInputProvider` provides a list of the user's available `audio-input` devices, the currently selected device, and the error when selecting device failed.
```
Comment (devalevenkatesh) [inline src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:5]: "when selecting device failed"  -> "if device selection fails"?

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx [hunk 3]
```diff
@@ -15,6 +15,9 @@ The `AudioInputProvider` provides a list of the user's available `audio-input` d
 
   // The device ID of the currently selected device
   selectedDevice: string | null
+
+  // The error when selecting device failed
+  selectDeviceError: Error | null
```
Comment (devalevenkatesh) [inline src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:20]: Mark it as optional here?

File: src/providers/MeetingProvider/MeetingManager.ts [hunk 4]
```diff
@@ -318,8 +324,8 @@ export class MeetingManager implements AudioVideoObserver {
         await this.audioVideo?.chooseAudioInputDevice(
           this.audioInputDevices[0].deviceId
         );
-      } catch (e) {
-        console.error(`Error in selecting audio input device - ${e}`);
+      } catch (error) {
```
Comment (devalevenkatesh) [inline src/providers/MeetingProvider/MeetingManager.ts:327]: Should we set and send the device selection error here as well?

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (xuesichao) [reply CHANGELOG.md:20]: Sure, should I change all of the 'Added' to 'Add'?

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx (hunk 2 above)
Comment (xuesichao) [reply src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:5]: Yes, this is much more clear! Will do!

File: CHANGELOG.md (hunk 1 above)
Comment (devalevenkatesh) [reply CHANGELOG.md:20]: Actually never mind, you can just change yours for now. We have a lot of "Added" elsewhere anyways 😅 .

File: src/providers/MeetingProvider/MeetingManager.ts (hunk 4 above)
Comment (xuesichao) [reply src/providers/MeetingProvider/MeetingManager.ts:327]: This is a good question. From my points, the issue is only about `selectXxxxxInputDeviceError()`, so I only change those functions. But generally, we may want to set and send the error wherever there is a calling to `audioVideo?.chooseXxxxxInputDevice()` API. How do you think? 
@ltrung could you give us some advice here?

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx (hunk 3 above)
Comment (xuesichao) [reply src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:20]: Yes, I should.

File: CHANGELOG.md (hunk 1 above)
Comment (xuesichao) [reply CHANGELOG.md:20]: I really want to fix all tense issues at a same time, lol

File: src/providers/MeetingProvider/MeetingManager.ts (hunk 4 above)
Comment (devalevenkatesh) [reply src/providers/MeetingProvider/MeetingManager.ts:327]: Yeah, I agree, we can scope this to currently reported issues only. I think no need to address this.

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx [hunk 5]
```diff
@@ -17,7 +17,7 @@ The `AudioInputProvider` provides a list of the user's available `audio-input` d
   selectedDevice: string | null
 
   // The error when selecting device failed
```
Comment (michhyun1) [inline src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:19]: nit: 'The error if selecting a device fails'

File: src/providers/DevicesProvider/docs/VideoInputProvider.stories.mdx [hunk 6]
```diff
@@ -17,7 +17,7 @@ Provides a list of the user's available `video-input` devices, the currently sel
   selectedDevice: string | null
   
   // The error when selecting device failed
```
Comment (michhyun1) [inline src/providers/DevicesProvider/docs/VideoInputProvider.stories.mdx:19]: nit: 'The error if selecting a device fails'

File: src/providers/DevicesProvider/docs/useAudioInputs.stories.mdx [hunk 7]
```diff
@@ -20,7 +20,7 @@ The `useAudioInputs` hook returns a list of the user's available `audio-input` d
   selectedDevice: string | null
 
   // The error when selecting device failed
```
Comment (michhyun1) [inline src/providers/DevicesProvider/docs/useAudioInputs.stories.mdx:22]: same here

File: src/providers/DevicesProvider/docs/useVideoInputs.stories.mdx [hunk 8]
```diff
@@ -20,7 +20,7 @@ The `useVideoInputs` hook returns a list of the user's available `video-input` d
   selectedDevice: string | null
 
   // The error when selecting device failed
```
Comment (michhyun1) [inline src/providers/DevicesProvider/docs/useVideoInputs.stories.mdx:22]: same here

File: src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx (hunk 5 above)
Comment (xuesichao) [reply src/providers/DevicesProvider/docs/AudioInputProvider.stories.mdx:19]: Oh, yes, I forget here, thanks Michael

**Comment by ltrung** [comment]
@xuesichao Can you squash your commit into one?

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]
I think during squashing you ended up adding my last [PR](https://github.com/aws/amazon-chime-sdk-component-library-react/pull/491) in this commit. Could you remove that 😅 ?

**Comment by xuesichao** [comment]
> @xuesichao Can you squash your commit into one?

@ltrung done

**Comment by xuesichao** [comment]
> I think during squashing you ended up adding my last [PR](https://github.com/aws/amazon-chime-sdk-component-library-react/pull/491) in this commit. Could you remove that 😅 ?

Thanks for your help, fixed 😁.

**Review by ltrung — APPROVED** [review]

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**